cartography/intel/aws/s3.py

```diff
@@ -102,21 +91,45 @@ def get_acl(bucket: Dict, client: botocore.client.BaseClient) -> Optional[str]:
     try:
         acl = client.get_bucket_acl(Bucket=bucket['Name'])
     except ClientError as e:
-        if "AccessDenied" in e.args[0]:
-            logger.warning("Failed to retrieve S3 bucket {} ACL - Access Denied".format(bucket['Name']))
+        if _is_common_exception(e, bucket):
             return None
-        elif "NoSuchBucket" in e.args[0]:
-            logger.warning("Failed to retrieve S3 bucket {} ACL - No Such Bucket".format(bucket['Name']))
-            return None
-        elif "AllAccessDisabled" in e.args[0]:
-            logger.warning("Failed to retrieve S3 bucket {} ACL - Bucket is disabled".format(bucket['Name']))
+        else:
+            raise
+    return acl
+
+
+@timeit
+def get_encryption(bucket: Dict, client: botocore.client.BaseClient) -> Optional[str]:
+    """
+    Gets the S3 bucket default encryption configuration. Returns encryption configuration or None if not enabled
+    """
+    try:
+        encryption = client.get_bucket_encryption(Bucket=bucket['Name'])
```

Comment:
Just wondering, is there an AWS API that returns these all in batch? Like can we say "get me all the bucket encryption objects" and then it'll return a list of dicts for all buckets in our environment?

Reply:
Alas, you have to make a call per bucket :(